torch/csrc/jit/passes/onnx/shape_type_inference.cpp

```diff
@@ -1992,15 +1984,19 @@ void ONNXShapeTypeInference(
       //       The conversion here is incomplete for these ops.
       //       e.g: ListConstruct, ListUnpack, etc.
       std::shared_ptr<onnx::ModelProto> model_proto;
-      SymbolDimMap symbol_map;
-      ConvertGraphToONNXProto(n_graph, model_proto, symbol_map, opset_version);
+      ConvertGraphToONNXProto(n_graph, model_proto, symbol_dim_map, opset_version);
       GRAPH_DEBUG(
           "ONNX graph to run shape inference: ", prettyPrint(*model_proto));
 
       // infer shape
       try {
-        onnx::shape_inference::InferShapes(*model_proto);
-        UpdateOutputTypeByONNXProto(n, clone_node, *model_proto, symbol_map);
+        // TODO: add data propagation supports for more operators
+        if (n->kind() == ::c10::onnx::Shape || n->kind() == ::c10::onnx::Gather) {
```

Comment:
would it make sense to write as switch statement, and leave default for the below `InferShapes`?